src/network/networkd-ndisc.c

```diff
@@ -13,12 +13,71 @@
 #include "networkd-manager.h"
 #include "networkd-ndisc.h"
 #include "networkd-route.h"
+#include "string-util.h"
 #include "strv.h"
 
 #define NDISC_DNSSL_MAX 64U
 #define NDISC_RDNSS_MAX 64U
 #define NDISC_PREFIX_LFT_MIN 7200U
 
+#define DAD_CONFLICTS_IDGEN_RETRIES_RFC7217 3
+
+/* https://tools.ietf.org/html/rfc5453 */
+/* https://www.iana.org/assignments/ipv6-interface-ids/ipv6-interface-ids.xml */
+
+#define SUBNET_ROUTER_ANYCAST_ADDRESS_RFC4291 ((struct in6_addr) { .s6_addr = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 } })
+#define RESERVED_IPV6_INTERFACE_IDENTIFIERS_ADDRESS_RFC4291 ((struct in6_addr) { .s6_addr = { 0x02, 0x00, 0x5E, 0xFF, 0xFE } })
+#define RESERVED_SUBNET_ANYCAST_ADDRESSES_RFC4291 ((struct in6_addr) { .s6_addr = { 0xFD, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF } })
+
+#define NDISC_APP_ID SD_ID128_MAKE(13,ac,81,a7,d5,3f,49,78,92,79,5d,0c,29,3a,bc,7e)
+
+static bool ipv6_interface_identifiers(const struct in6_addr *addr) {
+
+        assert(addr);
+
+        if (!memcmp(addr, &SUBNET_ROUTER_ANYCAST_ADDRESS_RFC4291, 8))
+                return true;
+
+        if (!memcmp(addr, &RESERVED_IPV6_INTERFACE_IDENTIFIERS_ADDRESS_RFC4291, 5))
+                return true;
+
+        if (!memcmp(addr, &RESERVED_SUBNET_ANYCAST_ADDRESSES_RFC4291, 7))
+                return true;
+
+        return false;
+}
+
+static int ipv6_makestableprivate(Link *link, const struct in6_addr *prefix, int prefix_len, uint8_t dad_counter, struct in6_addr *addr) {
```

Comment:
we use all kinds of different types for IP address prefix lengths, but it's in most cases an unsigned type, i.e. unsigned or uint8_t. I think we should do the same here, i.e. use "unsigned"? that should make the (size_t) cast in the MAX/DIV_ROUND_UP line below unnecessary even.